metricbeat/module/kubernetes/state_deployment/state_deployment_test.go

```diff
@@ -56,24 +56,27 @@ func TestEventMapping(t *testing.T) {
 		"hosts":      []string{server.URL},
 	}
 
-	f := mbtest.NewEventsFetcher(t, config)
-
-	events, err := f.Fetch()
-	assert.NoError(t, err)
+	f := mbtest.NewReportingMetricSetV2(t, config)
+	events, errs := mbtest.ReportingFetchV2(f)
+	if len(errs) > 0 {
+		t.Fatalf("Expected 0 error, had %d. %v\n", len(errs), errs)
+	}
+	assert.NotEmpty(t, events)
 
 	assert.Equal(t, 5, len(events), "Wrong number of returned events")
 
 	testCases := testCases()
 	for _, event := range events {
-		name, err := event.GetValue("name")
+		metricsetFields := event.MetricSetFields
+		name, err := metricsetFields.GetValue("name")
 		if err == nil {
-			namespace, err := event.GetValue("_module.namespace")
+			namespace, err := metricsetFields.GetValue("_module.namespace")
```

Comment:
I'd like to see a whole event for this metricset, with reporter V2 we shouldn't have these `_module.namespace` fields, I think they were a trick in the old reporter to set the "namespace" when you want to do it different to the module name.

I mean, without setting this namespace, the fields in this module would be placed under `kubernetes.state_deployment`, but they are being set under `kubernetes.deployment`. We have to check if this continues working with reporter V2. We may need to set `mb.Event.Namespace` instead.